File: src/main/java/me/lokka30/levelledmobs/compatibility/Compat1_19.java
package me.lokka30.levelledmobs.compatibility;

import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;
import java.util.stream.Collectors;
import java.util.stream.Stream;
import org.bukkit.entity.EntityType;
import org.jetbrains.annotations.NotNull;

public class Compat1_19 {

    public static HashSet<EntityType> getPassiveMobs() {
        return Stream.of(
            EntityType.valueOf("ALLAY"),
            EntityType.valueOf("TADPOLE"),
            EntityType.valueOf("FROG")
        ).collect(Collectors.toCollection(HashSet::new));
    }

    public static HashSet<EntityType> getHostileMobs() {
        return Stream.of(
            EntityType.valueOf("WARDEN")
        ).collect(Collectors.toCollection(HashSet::new));
    }

    public static HashSet<EntityType> getAquaticMobs() {
        return Stream.of(
            EntityType.valueOf("TADPOLE")
        ).collect(Collectors.toCollection(HashSet::new));
    }

    public static @NotNull Set<String> all19Mobs() {
        final Set<String> names = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
        names.addAll(List.of("ALLAY", "TADPOLE", "FROG", "WARDEN"));
        return names;
    }
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/compatibility/Compat1_19.kt
package io.github.arcaneplugins.levelledmobs.compatibility

import java.util.TreeSet
import org.bukkit.entity.EntityType

/**
 * Holds lists of entity types that are only present in minecraft 1.19 and newer.  Must be a
 * separate class to maintain compatibility with older versions
 *
 * @author stumper66
 * @since 3.6.0
 */
object Compat1_19 {
    fun getPassiveMobs(): MutableSet<EntityType> {
        return mutableSetOf(
            EntityType.ALLAY,
            EntityType.TADPOLE,
            EntityType.FROG
        )
    }

    fun getHostileMobs(): MutableSet<EntityType> {
        return mutableSetOf(
            EntityType.WARDEN
        )
    }

    fun getAquaticMobs(): MutableSet<EntityType> {
        return mutableSetOf(
            EntityType.TADPOLE
        )
    }

    fun all19Mobs(): MutableSet<String> {
        val names: MutableSet<String> = TreeSet(java.lang.String.CASE_INSENSITIVE_ORDER)
        names.addAll(listOf("ALLAY", "TADPOLE", "FROG", "WARDEN"))
        return names
    }
}